controllers/volumereplicationgroup_controller.go

```diff
@@ -1018,6 +1021,31 @@ func (v *VRGInstance) updateVRGClusterDataProtectedCondition() {
 	}
 }
 
+func (v *VRGInstance) updateVRGLastSyncTime() {
+	var leastLastSyncTime *metav1.Time
+
+	for _, protectedPVC := range v.instance.Status.ProtectedPVCs {
+		// If any protected PVC reports nil, report that back (no sync time available)
+		if protectedPVC.LastSyncTime == nil {
+			leastLastSyncTime = nil
+
+			break
+		}
+
+		if leastLastSyncTime == nil {
+			leastLastSyncTime = protectedPVC.LastSyncTime
+
+			continue
+		}
+
+		if protectedPVC.LastSyncTime != nil && protectedPVC.LastSyncTime.Before(leastLastSyncTime) {
```

Comment:
If I understand this correctly, given the following:

PVC1: LastSyncTime = 2020
PVC2: LastSyncTime = 2021
PVC3: LastSyncTime = 2019
The VRG lastSyncTime then should be 2021 right? If the answer is yes, then isn't this line going to set the leastLastSyncTime to 2019? And then we end up with 2019 for the VRG, instead of 2021?

I mean line 1046 will preserve the least last sync time as the name indicates, but should be the latest?

Reply:
We report the least time, as the volume that is most lagging determines how far back the group is. If there are 3 volumes as above, then as a group they are as far back as the 2019 volume.

Reply:
I see. That makes sense now.